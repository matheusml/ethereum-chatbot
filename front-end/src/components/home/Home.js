import React from "react";
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableOpacity
} from "react-native";

import dismissKeyboard from "dismissKeyboard";

import Chat from "../chat/Chat";

export default class Home extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      channel: ""
    };

    this.enterChat = this.enterChat.bind(this);
  }

  enterChat() {
    const { navigate } = this.props.navigation;
    navigate("Chat", { channel: this.state.channel });

    dismissKeyboard();
  }

  render() {
    return (
      <View style={styles.container}>
        <TextInput
          style={styles.textInput}
          placeholder={"Channel name"}
          returnKeyType="next"
          autoCapitalize="none"
          onChangeText={channel => this.setState({ channel })}
          value={this.state.channel}
          underlineColorAndroid="transparent"
          multiline
        />
        <TouchableOpacity style={styles.button} onPress={this.enterChat}>
          <Text style={styles.buttonTitle}>Enter</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#ffffff"
  },
  textInput: {
    width: 110,
    margin: 10,
    padding: 3,
    fontSize: 16
  },
  button: {
    backgroundColor: "lightseagreen",
    height: 40,
    margin: 10,
    width: 100,
    borderRadius: 5,
    padding: 3,
    alignItems: "center",
    justifyContent: "center"
  },
  buttonTitle: {
    color: "#ffffff",
    fontSize: 18,
    fontWeight: "bold"
  }
});
