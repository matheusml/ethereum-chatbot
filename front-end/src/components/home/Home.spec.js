import React from "react";
import dismissKeyboard from "dismissKeyboard";

import Home from "./Home";

import renderer from "react-test-renderer";

jest.mock("dismissKeyboard");

const navigateMock = jest.fn();

const props = {
  navigation: {
    navigate: navigateMock
  }
};

describe("Home", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<Home {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should have an initial state", () => {
    const instance = renderer.create(<Home {...props} />).getInstance();
    expect(instance.state).toEqual({ channel: "" });
  });

  describe("enterChat", () => {
    it("should invoke navigate", () => {
      const instance = renderer.create(<Home {...props} />).getInstance();
      instance.enterChat();
      expect(navigateMock).toHaveBeenCalled();
    });

    it("should invoke dismissKeyboard", () => {
      const instance = renderer.create(<Home {...props} />).getInstance();
      instance.enterChat();
      expect(dismissKeyboard).toHaveBeenCalled();
    });
  });
});
