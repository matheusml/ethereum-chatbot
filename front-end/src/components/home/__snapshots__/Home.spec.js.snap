// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    multiline={true}
    onChangeText={[Function]}
    placeholder="Channel name"
    returnKeyType="next"
    style={
      Object {
        "fontSize": 16,
        "margin": 10,
        "padding": 3,
        "width": 110,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightseagreen",
        "borderRadius": 5,
        "height": 40,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 3,
        "width": 100,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Enter
    </Text>
  </View>
</View>
`;
